src/app/event_handlers.py

```diff
@@ -11,10 +19,29 @@
 @webhook_handler.add(MessageEvent, message=TextMessage)
 def handle_text_message_event(event: MessageEvent):
     reply_token = event.reply_token
+    message = event.message.text
 
-    # default response
-    message = HelpMessageCreator().generate()
-    line_api.reply_message(reply_token, message)
+    if product_code := UqProductCodeParser(message).get_product_code_from_url():
+        reply_message = generate_response_for_given_product_code(product_code)
+    else:
+        reply_message = HelpMessageCreator().generate()
```

Comment:
We can try to find a better way for the dispatching logic. We need to modify this method in the current design whenever a new text message command is added. But we don't need to implement it in this PR as this might not be an easy-to-solve problem.